src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -256,7 +256,7 @@ public String produceDts(Depgraph depgraph) {
   private Set<String> getShadowedProvides(TreeSet<String> provides) {
     Set<String> shadowedProvides = new TreeSet<>();
     for (String provide : provides) {
-      if (!provides.subSet(provide + ".", provide + "\uFFFF").isEmpty()) {
+      if (!provides.subSet(provide + ".", provide + ".\uFFFF").isEmpty()) {
```

Comment:
I don't think this deserves a whole test of the type we have been writing. But maybe this is a good reason to add unittests, what do you think?